src/core/search/indices.cc

```diff
@@ -71,19 +71,21 @@ absl::flat_hash_set<string> NormalizeTags(string_view taglist, bool case_sensiti
 NumericIndex::NumericIndex(PMR_NS::memory_resource* mr) : entries_{mr} {
 }
 
+bool NumericIndex::Matches(DocId id, DocumentAccessor* doc, string_view field) {
+  auto strings = doc->GetStrings(field);
+  return std::all_of(strings.begin(), strings.end(),
```

Comment:
so here you test that all field values are indexable? why all of them?

Reply:
Actually the result of GetStrings should be only one value

Reply:
From Redis:
```
127.0.0.1:6379> json.set j1 . '{"arr":[{"value":1},{"value":"one"}]}'
OK
127.0.0.1:6379> json.set j2 . '{"arr":[{"value":1},{"value":2}]}'
OK
127.0.0.1:6379> ft.create index on json schema $.arr[*].value AS value NUMERIC
OK
127.0.0.1:6379> ft.search index "*"
1) (integer) 1
2) "j2"
3) 1) "$"
   2) "{\"arr\":[{\"value\":1},{\"value\":2}]}"
```

Reply:
So all fields should have the same field type as index. Only then this document can be added

Reply:
I also see that we had another logic for the sort indices. We just parsed only the first value. It is a bug. But there is another problem that we are saving the sortable values